### PR TITLE
Bump version to Mattermost 7.3.0

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
Bump Mattermost Server version to pull the latest plugin api docs.

The only updated page is `extend/plugins/server/reference/`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,69 +1,69 @@
 module mattermost-developer-documentation
 
-go 1.17
+go 1.18
 
 require (
-	github.com/mattermost/mattermost-server/v6 v6.3.0
+	// mmgoget: github.com/mattermost/mattermost-server/v6@v7.3.0 is replaced by -> github.com/mattermost/mattermost-server/v6@5d1fe271e6
+	github.com/mattermost/mattermost-server/v6 v6.0.0-20220914120725-5d1fe271e68c
```
Comment (hanzei) [inline go.mod:7]: See https://community.mattermost.com/core/pl/xq1jy5pw8ibrmpx9uwswusozey for more context on why a commit is used instead of a version.

**Comment by neflyte** [comment]
The `go.mod` changes look okay and the docs build without error.
The changes to the build pipeline cannot be tested in a branch other than `master`, so we won't know if the changes work until this PR merges.

**Review by neflyte — APPROVED** [review]

**Comment by neflyte** [comment]
Adding @spirosoik to review the build pipeline changes.

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -3,7 +3,7 @@
 variables:
   AWS_DEFAULT_REGION: "us-east-1"
 
-  IMAGE_BUILD: cimg/go:1.17-node
+  IMAGE_BUILD: cimg/go:1.18-node
```
Comment (spirosoik) [inline .gitlab-ci.yml:6]: it is a good practice to pin to a specific digest sha of the image build

**Comment by spirosoik** [comment]
@neflyte for future reference just use @mattermost/core-build-engineers so someone can see it fast enough

File: .gitlab-ci.yml (hunk 2 above)
Comment (hanzei) [reply .gitlab-ci.yml:6]: Is it really in this case? Go only releases security updates as patch releases, so we want to have them every time.

Comment (spirosoik) [reply .gitlab-ci.yml:6]: Not super critical for this repo but keep in mind that if we pin the dependencies to an exact version in our manifest files, no matter when you build your code, the same top-level dependencies should be pulled in, making our builds more transparent and consistent. This helps a bit in terms of the supply chain security

**Review by spirosoik — APPROVED** [review]

**Pull request merged**